apps/src/templates/curriculumCatalog/noSectionsToAssignDialogs.jsx

```diff
@@ -6,27 +6,66 @@ import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
 import Typography from '@cdo/apps/componentLibrary/typography';
 import style from './no_sections_to_assign_dialog.module.scss';
 
+const signInURL = `/users/sign_in?user_return_to=${location.pathname}`;
+const upgradeAccountArticleURL =
+  'https://support.code.org/hc/en-us/articles/360023222371-How-can-I-change-my-account-type-from-student-to-teacher-or-vice-versa';
+
 export const SignInToAssignSectionsDialog = ({onClose}) => (
   <NoSectionsToAssignBaseDialog
     headerText={i18n.signInToAssign()}
     helpText={i18n.signInToAssignHelpText()}
     onClose={onClose}
     buttonText={i18n.signInOrCreateAccount()}
-    href={`/users/sign_in?user_return_to=${location.pathname}`}
+    href={signInURL}
   />
 );
 
 SignInToAssignSectionsDialog.propTypes = {
   onClose: PropTypes.func.isRequired,
 };
 
+export const UpgradeAccountToAssignSectionsDialog = ({onClose}) => (
+  <NoSectionsToAssignBaseDialog
+    headerText={i18n.upgradeAccountToAssign()}
+    helpText={i18n.upgradeAccountToAssignHelpText()}
+    onClose={onClose}
+    buttonText={i18n.upgradeAccountToAssignButtonText()}
+    href={upgradeAccountArticleURL}
+  />
+);
+
+UpgradeAccountToAssignSectionsDialog.propTypes = {
+  onClose: PropTypes.func.isRequired,
+};
+
+export const CreateSectionsToAssignSectionsDialog = ({onClose, onClick}) => (
+  <NoSectionsToAssignBaseDialog
+    headerText={i18n.createClassSectionsToAssign()}
+    helpText={i18n.createClassSectionsToAssignHelpText()}
+    onClose={onClose}
+    buttonText={i18n.createClassSectionToAssignButton()}
+    onClick={onClick}
+  />
+);
+
+CreateSectionsToAssignSectionsDialog.propTypes = {
+  onClose: PropTypes.func.isRequired,
+  onClick: PropTypes.func.isRequired,
+};
+
 const NoSectionsToAssignBaseDialog = ({
   headerText,
   helpText,
   onClose,
   buttonText,
   href,
+  onClick,
+  openInNewTab,
 }) => {
+  if (onClick ^ href) {
+    throw new Error('Expect exactly one of onClick or href');
+  }
```

Comment:
Wouldn't xor here only return true if there is exactly one of onClick or href, not the opposite?

Reply:
Ah, I had it backwards! But now the tests are failing? Investigating ... thanks for raising

Reply:
Oh yikes, turns out Javascript's XOR is only for boolean values 🤯 . So this was doing nothing. Updated to coerce these into Boolean values. Thank you for catching this!